### PR TITLE
Ants can't traumatize ghosts, ants show up on top of pipes + Ant QOL

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Adds a check to make sure that the person being covered in ants is conscious before they begin to scream from being covered in a shitload of ants.

Pouring ants on pipes, vents, and cables will work just like pouring them on an empty tile now, to make it so you don't need to pixel-search to place down a mound of ants successfully.

## Why It's Good For The Game
Makes it easier to pour ants & stops people from being covered in ants and screaming in the afterlife because of it.

## Changelog

:cl: Wallem
qol: Ants have become easier to pour in tiles with vents, pipes or cables.
fix: Makes ants incapable of affecting the afterlife.
fix: Ants, gibs, and garbage cleanable decals now properly show up above pipes.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
ants

**Comment by Wallemations** [comment]
ants

**Comment by Iamgoofball** [comment]
ants

**Comment by itseasytosee** [comment]
image

**Comment by Wallemations** [comment]
After being reminded that planes exist, I fixed #62495 as well.

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -980,19 +980,21 @@
...
 
 /datum/status_effect/ants/refresh(effect, amount_left)
 	var/mob/living/carbon/human/victim = owner
-	if(isnum(amount_left) && ants_remaining >= 1)
-		if(!prob(1)) // 99%
-			to_chat(victim, "<span class='userdanger'>You're covered in MORE ants!</span>")
-		else // 1%
-			victim.say("AAHH! THIS SITUATION HAS ONLY BEEN MADE WORSE WITH THE ADDITION OF YET MORE ANTS!!", forced = /datum/status_effect/ants)
+	if(isnum(amount_left) && ants_remaining >= 1 && victim.stat >= HARD_CRIT)
+		if(victim.stat <= SOFT_CRIT) // Unconcious people won't get messages
```
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:993]:
```
if victim.stat >= 3
   if victim.stat <= 1
```

?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/debuffs.dm (hunk 1 above)
Comment (Wallemations) [reply code/datums/status_effects/debuffs.dm:993]: Oh good catch, that's supposed to have a !

Comment (Wallemations) [reply code/datums/status_effects/debuffs.dm:993]: Wait no, I had it right the first time, that's correct.

Comment (Wallemations) [reply code/datums/status_effects/debuffs.dm:993]: I just realized soft crit isn't unconscious, fixing that

Comment (Wallemations) [reply code/datums/status_effects/debuffs.dm]: There we go, fixed it.

Comment (ShizCalev) [reply code/datums/status_effects/debuffs.dm]:
```
if(victim.stat >= HARD_CRIT)
		if(victim.stat <= UNCONSCIOUS)
```
HARD_CRIT = 3
UNCONSCIOUS = 2

written out, this is
```
if(victim.stat is greater than or equal to 3)
		if(victim.stat is less than or equal to 2)    <----- you can't reach this is victim's is 3+
```

File: code/datums/status_effects/debuffs.dm [hunk 2]
```diff
@@ -980,19 +980,21 @@
...
 
 /datum/status_effect/ants/refresh(effect, amount_left)
 	var/mob/living/carbon/human/victim = owner
-	if(isnum(amount_left) && ants_remaining >= 1)
-		if(!prob(1)) // 99%
-			to_chat(victim, "<span class='userdanger'>You're covered in MORE ants!</span>")
-		else // 1%
-			victim.say("AAHH! THIS SITUATION HAS ONLY BEEN MADE WORSE WITH THE ADDITION OF YET MORE ANTS!!", forced = /datum/status_effect/ants)
+	if(isnum(amount_left) && ants_remaining >= 1 && victim.stat >= HARD_CRIT)
+		if(victim.stat <= UNCONSCIOUS) // Unconcious people won't get messages
```
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:993]:
```
if(victim.stat >= HARD_CRIT)
		if(victim.stat <= UNCONSCIOUS)
```
HARD_CRIT = 3
UNCONSCIOUS = 2

written out, this is
```
if(victim.stat is greater than or equal to 3)
		if(victim.stat is less than or equal to 2)    <----- you can't reach this is victim's is 3+
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/debuffs.dm (hunk 2 above)
Comment (Wallemations) [reply code/datums/status_effects/debuffs.dm:993]: Goddamnit I knew I was messing this up
For some reason I can't get my head around the concept that the lower the number the more conscious you are, I keep thinking it's the other way around

Comment (Wallemations) [reply code/datums/status_effects/debuffs.dm]: I _hope_ I did that right this time

File: code/datums/status_effects/debuffs.dm [hunk 3]
```diff
@@ -980,19 +980,21 @@
 	var/ants_remaining = 0
 
 /datum/status_effect/ants/on_creation(mob/living/new_owner, amount_left)
-	if(isnum(amount_left))
-		to_chat(new_owner, "<span class='userdanger'>You're covered in ants!</span>")
+	if(isnum(amount_left) && new_owner.stat <= HARD_CRIT)
+		if(new_owner.stat <= UNCONSCIOUS) // Unconcious people won't get messages
```
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:984]: <

Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:984]: <

File: code/datums/status_effects/debuffs.dm [hunk 4]
```diff
@@ -980,19 +980,21 @@
 	var/ants_remaining = 0
 
 /datum/status_effect/ants/on_creation(mob/living/new_owner, amount_left)
-	if(isnum(amount_left))
-		to_chat(new_owner, "<span class='userdanger'>You're covered in ants!</span>")
+	if(isnum(amount_left) && new_owner.stat <= HARD_CRIT)
```
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:983]: should just be <

File: code/datums/status_effects/debuffs.dm (hunk 3 above)
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:984]: should just be <

File: code/datums/status_effects/debuffs.dm [hunk 5]
```diff
@@ -980,19 +980,21 @@
...
 	. = ..()
 
 /datum/status_effect/ants/refresh(effect, amount_left)
 	var/mob/living/carbon/human/victim = owner
-	if(isnum(amount_left) && ants_remaining >= 1)
-		if(!prob(1)) // 99%
-			to_chat(victim, "<span class='userdanger'>You're covered in MORE ants!</span>")
-		else // 1%
-			victim.say("AAHH! THIS SITUATION HAS ONLY BEEN MADE WORSE WITH THE ADDITION OF YET MORE ANTS!!", forced = /datum/status_effect/ants)
+	if(isnum(amount_left) && ants_remaining >= 1 && victim.stat <= HARD_CRIT)
```
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:992]: should just be <

File: code/datums/status_effects/debuffs.dm [hunk 6]
```diff
@@ -980,19 +980,21 @@
...
+		if(victim.stat <= UNCONSCIOUS) // Unconcious people won't get messages
```
Comment (ShizCalev) [inline code/datums/status_effects/debuffs.dm:993]: should just be <

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Wallemations** [comment]
No those are all right, it's <= because it's also if it equals it

**Pull request merged**